sql/plan/alter_foreign_key.go

```diff
@@ -0,0 +1,168 @@
+package plan
+
+import (
+	"fmt"
+	"github.com/liquidata-inc/go-mysql-server/sql"
+	"gopkg.in/src-d/go-errors.v1"
+	"strings"
+)
+
+var (
+	// ErrNoForeignKeyAlterSupport is returned when the table does not support ALTER TABLE ... FOREIGN KEY operations.
+	ErrNoForeignKeyAlterSupport = errors.NewKind("the table does not support foreign key alter operations")
+	// ErrForeignKeyMissingColumns is returned when an ALTER TABLE ADD FOREIGN KEY statement does not provide any columns
+	ErrForeignKeyMissingColumns = errors.NewKind("cannot create a foreign key without columns")
+	// ErrForeignKeyActionNotImplemented is returned when the action has not been implemented
+	ErrForeignKeyActionNotImplemented = errors.NewKind("alter table foreign key action is not implemented: %v")
+	// ErrAddForeignKeyNonExistentColumn is returned when a key is provided in the foreign key that isn't in the table
+	ErrAddForeignKeyNonExistentColumn = errors.NewKind("column `%v` does not exist in the table")
+	// ErrAddForeignKeyDuplicateColumn is returned when an ALTER TABLE ADD FOREIGN KEY statement has the same column multiple times
+	ErrAddForeignKeyDuplicateColumn = errors.NewKind("cannot have duplicates of columns in a foreign key: `%v`")
+)
+
+type ForeignKeyAction byte
+const (
+	ForeignKeyAction_Add ForeignKeyAction = iota
+	ForeignKeyAction_Drop
+)
+
+type AlterForeignKey struct {
+	Action  ForeignKeyAction
+	Table   sql.Node
```

Comment:
Consider embedding UnaryNode instead of doing a custom version of the same thing